freqtrade/exchange/kraken.py

```diff
@@ -2,11 +2,42 @@
 import logging
 from typing import Dict
 
+import ccxt
+
+from freqtrade import OperationalException, TemporaryError
 from freqtrade.exchange import Exchange
+from freqtrade.exchange.exchange import retrier
 
 logger = logging.getLogger(__name__)
 
 
 class Kraken(Exchange):
 
     _params: Dict = {"trading_agreement": "agree"}
+
+    @retrier
+    def get_balances(self) -> dict:
+        if self._config['dry_run']:
+            return {}
+
+        try:
+            balances = self._api.fetch_balance()
+            # Remove additional info from ccxt results
+            balances.pop("info", None)
+            balances.pop("free", None)
+            balances.pop("total", None)
+            balances.pop("used", None)
+
+            orders = self._api.fetch_open_orders()
+            order_list = [[x["symbol"].split("/")[0 if x["side"] == "sell" else 1],
+                           x["remaining"], x["side"], x["amount"], ] for x in orders]
```

Comment:
* Should we store x["side"], x["amount"] here? Used for debugging?
* A list of tuples may be more effective?

Reply:
My apologies. I don't have enough with python to test this on my own. If anyone is will to send me a step by step. I'd be more than willing to run some tests.

Reply:
You don't have to test - but you were the one raising the issue about kraken balance not working, so i thought you may want to test this version.

You should be able to run the following git commands to test this branch if you're interrested:

```
git remote -v 
# verify if origin is really pointing to github.com/freqtrade/freqtrade, otherwise replace origin with the name you see above
git fetch origin
git checkout -b kraken_balance origin/kraken_balance 

### Do tests

# Move back to develop / master
git checkout develop
# git checkout master
```

Reply:
> A list of tuples may be more effective?

yes it would be - but if list vs. tuple becomes an issue in this function i'm more than happy... :smile: 
my real worry is exchange delays due to rate-limiting within `fetch_open_orders()` ...

While i didn't experience this while testing, it's something to keep in mind.